### DISCUSSION TITLE
WiFi connection on MacOS

### DISCUSSION THREAD

**Discussion by RiSox22** [cedricp/ddt4all#899] [Q&A]
Hello, I use ddt4all v2.1.1 on macOS 13.5.1.
It works well with an ELM USB plug.
But with a WiFi plug (KingBolen v1.5) impossible. Yet the ELM plug works with EOBD Facile from Mac App Store. Why ?
I'm stuck on "Attempting to open port192.168.0.10:35000 @ 38400" => no error message, just freeze

**Comment by KarelSvo** [comment]
https://github.com/cedricp/ddt4all/pull/659/commits/1c68148f3e1eb0102878b4fd20acdad54bd1c401

**Comment by RiSox22** [comment]
Thks @KarelSvo it’s seems to work now, but it’s very very slow : >10min for Elm test
Only few seconds in USB
Normal ? And why the Line isn’t commit ?

![image](https://github.com/cedricp/ddt4all/assets/32896210/1011a1ae-8052-41c8-b8f5-605ab9e8bbf3)

![image](https://github.com/cedricp/ddt4all/assets/32896210/068a453f-5b6f-4d69-9c72-1cd587310ceb)

**Comment by KarelSvo** [comment]
I prefer to use the stable Python 2 version. The Python 3 version still contains too many bugs.

**Reply by RiSox22** [reply]
ddt4all v2.1.1 work with python2 ?

**Comment by KarelSvo** [comment]
https://github.com/cedricp/ddt4all/releases/tag/1.5

**Reply by RiSox22** [reply]
any change possible in the .py to increase speed / decrease lag ? in v2.1.1
I don't have py2

**Comment by KarelSvo** [comment]
https://github.com/cedricp/ddt4all/tree/34ce895d656e2c39e8450692d19679d2ec848e62


After that, it was simply changed to Python 3 instead of opening a new branch for it.

https://github.com/cedricp/ddt4all/commits/master?after=0868a182f2cb9e2326c94b4a2b150909db71bd96+209&branch=master&qualified_name=refs%2Fheads%2Fmaster

The latest py2 + QT4 version
https://github.com/cedricp/ddt4all/tree/d901ddf94b677493816098a364a0fd5ba280c1af

**Comment by RiSox22** [comment]
I've installed Python v2.7.18 on macOs but for PyQt it's more complicated... :/  PyQt4 is outdated and not available on pip. 
PyQt5, not easy . I was install it from brew with py3 and ddt4all v2.1.1

<img width="1242" alt="image" src="https://github.com/cedricp/ddt4all/assets/32896210/b878303c-970e-4b7d-90f0-bf1fa6e03b4d">
